### ISSUE TITLE
Copper bars "demo" not working right

### ISSUE THREAD

**Issue by aybe** [joncampbell123/dosbox-x#5113]
### Describe the bug

30 years ago, I bought this book: 🤣

![301653922](https://github.com/joncampbell123/dosbox-x/assets/1537591/b321f27e-41b7-4478-8184-02690bb5a040)

TTF output didn't work but switching to surface finally shows something:

![image](https://github.com/joncampbell123/dosbox-x/assets/1537591/78a77490-65a0-43e4-8acd-8014eb8398a2)

But I vividly remember these bars would mix like so:

![2c8a1f65fadc33da6b7b104be2707d6c](https://github.com/joncampbell123/dosbox-x/assets/1537591/d7849926-6f04-4695-a6d7-ab980315006f)

As it was a programming book, the code was provided:

[COPPER.zip](https://github.com/user-attachments/files/16154409/COPPER.zip)

Just in case, the archived CD for the book:

https://www.abandonware-france.org/bibliotheque/livres/le-meilleur-programmation-jeux-3d-27/

Thanks! 😀

### Steps to reproduce the behaviour

Switch to 'surface' output, start `COPPER.EXE`.

### What operating system(s) this bug have occurred on?

Windows 11

### What version(s) of DOSBox-X have this bug?

2024.03.01

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
I don't think this demo intended for the red and green bars to additive blend. Notice how the bars are made to look like they are rotating around each other, switching whichever one is on top as they go.

The TTF output isn't meant for this kind of demo effect. It's intended to display text, that's it.

**Comment by aybe** [comment]
My bad, you are right...

But I didn't dream, went through all demos in the CD and it was another one!!!🥳:

![image](https://github.com/joncampbell123/dosbox-x/assets/1537591/8ccb8ab0-a4d1-47e8-a6e7-c00c924ceb8f)

![image](https://github.com/joncampbell123/dosbox-x/assets/1537591/2c724c3d-2af6-4619-b60f-678b6e0f8418)

[AARDVARK.zip](https://github.com/user-attachments/files/16169553/AARDVARK.zip)

Funny how memories can be at the same time accurate and not... 

---

Another one, it's supposed to wobble but it isn't:

![image](https://github.com/joncampbell123/dosbox-x/assets/1537591/2bd2a2a2-7075-49a4-81b6-114a88245f36)

[WOBBLER.zip](https://github.com/user-attachments/files/16169567/WOBBLER.zip)

According to the sources, I pressed a key once to get the effect started, waited, but nothing changed.

Other than that, the other demos in the CD do work quite well!

Thanks!!!

**Comment by joncampbell123** [comment]
The wobble one was probably coded for some SVGA chipsets that allow it (I know ATI chipsets do). Most SVGA chipsets don't support it.

The common technique is to play with the HPEL horizontal panning register per scanline. Most cards latch that at display start, so changes mid frame don't count. ATI chipsets let you change it at any time which means it takes effect next line.

The less common one, used by the 1992 demo "Copper" plays with the placement of the hsync pulse and depends on your analog CRT monitor's phase lock loop to make it wobble.

**Comment by joncampbell123** [comment]
S3 chipsets as far as I know don't allow changing HPEL per scanline.

To enable it anyway, add to your dosbox.conf:

    [video]
    allow hpel effects=true

**Comment by aybe** [comment]
I tried what you suggested but that didn't change anything:

```
[dosbox]
machine=svga_s3

[sdl]
output=surface

[video]
allow hpel effects=true

[cpu]
cycles=20000

[render]
scaler=normal2x forced
```
